lib/spritesheet.dart

```diff
@@ -6,66 +6,95 @@ import 'sprite.dart';
 import 'sprite_animation.dart';
 import 'extensions/vector2.dart';
 
-/// Utility class to help extract animations and sprites from a spritesheet image
+/// Utility class to help extract animations and sprites from a sprite sheet image.
+///
+/// A sprite sheet is a single image in which several regions can be defined as individual sprites.
+/// For the purposes of this class, all of these regions must be identically sized rectangles.
+/// You can use the [Sprite] class directly if you want to have varying shapes.
+///
+/// Each sprite in this sheet can be identified either by it's (row, col) pair or
+/// by it's "id", which is basically it's sequenced index if the image is put in a
+/// single line. The sprites can be used to compose an animation easily if they
+/// all the frames happen to be sequentially on the same row.
+/// Sprites are lazily generated but cached.
 class SpriteSheet {
-  int textureWidth;
-  int textureHeight;
-  int columns;
-  int rows;
+  /// The src image from which each sprite will be generated.
+  final Image image;
 
-  List<List<Sprite>> _sprites;
+  /// The size of each rectangle within the image that define each sprite.
+  ///
+  /// For example, if this sprite sheet is a tile map, this would be the tile size.
+  /// If it's an animation sheet, this would be the frame size.
+  final Vector2 srcSize;
+
+  final Map<int, Sprite> _spriteCache = {};
```

Comment:
You could use the MemoryCache class here: https://github.com/flame-engine/flame/blob/v1.0.0/lib/memory_cache.dart

Reply:
I don't think it fits here because the only thing it adds over the map is eviction, that we don't want on this case.